base/linalg/symmetric.jl

```diff
@@ -117,8 +117,16 @@ A_mul_B!{T<:BlasComplex,S<:StridedMatrix}(C::StridedMatrix{T}, A::StridedMatrix{
 *(A::HermOrSym, B::HermOrSym) = full(A)*full(B)
 *(A::StridedMatrix, B::HermOrSym) = A*full(B)
 
-factorize(A::HermOrSym) = bkfact(A.data, symbol(A.uplo), issym(A))
+bkfact(A::HermOrSym) = bkfact(A.data, symbol(A.uplo), issym(A))
+factorize(A::HermOrSym) = bkfact(A)
```

Comment:
This type of check is already happening in [`factorize`](https://github.com/JuliaLang/julia/blob/7affd0ced06c3596680a7ad69081aa7b98ba6ef9/base/linalg/dense.jl#L770-L781)? Since `factorize` is a general method which essentially should return the "best" factorization I think the same thing should apply if you have a `Hermitian` input matrix?